### ISSUE TITLE
vmstorage: process hanged after entering readonly mode

### ISSUE THREAD

**Issue by hagen1778** [VictoriaMetrics/VictoriaMetrics#2762]
**Describe the bug**
vmstorage process gone into a bad state after entering RO and didn't respond to SIGTERM.

**Expected behavior**
Entering RO should be safe and process should remain responsive.

**Logs**
```
panic: FATAL: cannot flush inmemory parts: cannot merge 2 inmemory parts: cannot merge 2 parts: cannot create destination part "<redacted: data dir path>/data/small/2022_06/tmp/16F871C992AA4CF2": cannot create directory "<redacted: data dir path>/data/small/2022_06/tmp/16F871C992AA4CF2": mkdir <redacted: data dir path>/data/small/2022_06/tmp/16F871C992AA4CF2: no space left on device
2022-06-17T14:35:03.622Z	panic	VictoriaMetrics/lib/storage/partition.go:755	FATAL: cannot flush inmemory parts: cannot merge 2 inmemory parts: cannot merge 2 parts: cannot create destination part "<redacted: data dir path>/data/small/2022_06/tmp/16F871C992AA4CF2": cannot create directory "<redacted: data dir path>/data/small/2022_06/tmp/16F871C992AA4CF2": mkdir <redacted: data dir path>/data/small/2022_06/tmp/16F871C992AA4CF2: no space left on device
2022-06-17T14:35:03.622Z	warn	VictoriaMetrics/lib/storage/storage.go:625	switching the storage at <redacted: data dir path> to read-only mode, since it has less than -storage.minFreeDiskSpaceBytes=10000000 of free space: 20480 bytes left
2022-06-17T14:35:03.604Z	warn	VictoriaMetrics/lib/storage/storage.go:625	switching the storage at <redacted: data dir path> to read-only mode, since it has less than -storage.minFreeDiskSpaceBytes=10000000 of free space: 20480 bytes left
```

**Version**
1.76.1

**Comment by valyala** [comment]
It looks like the issue is related to background merge, which can start when there is no free disk space - https://github.com/VictoriaMetrics/VictoriaMetrics/issues/2603 .

Temporary workaround: to increase `-storage.minFreeDiskSpaceBytes` to bigger values. By default it is set to 10000000 (10MB). This may be too low for high ingestion speeds. Try setting it to 100000000 (100MB) or even bigger values.

**Comment by valyala** [comment]
FYI, the issue must be addressed in the commit a0e7432e4248405440675bde3b98554f4aa73a81 , which has been included in [v1.81.0](https://docs.victoriametrics.com/CHANGELOG.html#v1810).
